File: Codigos estudiantes por lenguaje/PY/Luciano_Chavarria/Grafos/DFS.py
class Grafo:
    V = 0
    adj = [[],[],[],[],[],[]]
    def addEdge(self,v,w):
        self.adj[v].append(w)
    def DFSUtil(self,v,visited = []):
        visited[v] = True
        print(v," ")
        for i in range(len(self.adj[v])):
            n = self.adj[v][i]
            if not visited[n]:
                self.DFSUtil(n,visited)
    def DFS(self,v):
        visited = [False,False,False,False,False,False]
        self.DFSUtil(v,visited)

g = Grafo()
g.V=4
g.addEdge(0,1)
g.addEdge(0,2)
g.addEdge(1,2)
g.addEdge(2,0)
g.addEdge(2,3)
g.addEdge(3,3)

inicial = 0
print(f"Nodo inicial {inicial}")
g.DFS(inicial)

File: Codigos estudiantes por lenguaje/C++/Luciano_Chavarria/Grafos/DFS.cpp
#include <bits/stdc++.h>
#define MAX_V 101
//typedef pair Edge;
using namespace std;
vector<int> adj[MAX_V];

struct Graph {
    int V;
    void addEdge(int v, int w) {
        adj[v].push_back(w);
    }
    void DFSUtil(int v, bool visited[]) {
        visited[v] = true;
        cout << v << " ";
        for (int i = 0; i < adj[v].size(); ++i) {
            int n = adj[v][i];
            if (!visited[n]) {
                DFSUtil(n, visited);
            }
        }
    }
    void DFS(int v) {
        bool visited[V];
        DFSUtil(v, visited);
    }
};

int main() {
    Graph gra;
    gra.V=4;
    gra.addEdge(0, 1);
    gra.addEdge(0, 2);
    gra.addEdge(1, 2);
    gra.addEdge(2, 0);
    gra.addEdge(2, 3);
    gra.addEdge(3, 3);
    int initial = 0;
    gra.DFS(initial);
    return 0;
}
